operator/v1/types_ingress.go

```diff
@@ -483,6 +483,40 @@ type LoggingDestination struct {
 	Container *ContainerLoggingDestinationParameters `json:"container,omitempty"`
 }
 
+// IngressControllerCaptureHTTPHeader describes an HTTP header that should be
+// captured.
+type IngressControllerCaptureHTTPHeader struct {
+	// name specifies a header name.  Its value must be a valid HTTP header
+	// name as defined in RFC 2616 section 4.2.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^[\x21\x23-\x27\x2a\x2b\x2d\x2e\x30-\x39\x41-\x5a\x5e-\x7a\x7c\x7e]+$`
```

Comment:
why can't  you use characters?  these are all printable characters, right?

Reply:
It only matches valid header names, using a straightforward translation of the RFC definition to regexp:

       message-header = field-name ":" [ field-value ]
       field-name     = token

https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2

       token          = 1*<any CHAR except CTLs or separators>
       CHAR           = <any US-ASCII character (octets 0 - 127)>
       CTL            = <any US-ASCII control character
                        (octets 0 - 31) and DEL (127)>
       separators     = "(" | ")" | "<" | ">" | "@"
                      | "," | ";" | ":" | "\" | <">
                      | "/" | "[" | "]" | "?" | "="
                      | "{" | "}" | SP | HT
       SP             = <US-ASCII SP, space (32)>
       HT             = <US-ASCII HT, horizontal-tab (9)>

https://www.w3.org/Protocols/rfc2616/rfc2616-sec2.html#sec2.2

I guess I could use characters:
```
	// +kubebuilder:validation:Pattern=`^[-!#$%&'*+.0-9A-Z^_`a-z|~]+$`
```
Is that preferable?

Reply:
I went ahead and updated the pattern.